Source/Generic/ECLogManager.m

```diff
@@ -290,7 +293,11 @@ - (void)startup
 
 	[self loadSettings];
 	[self registerHandlers];
-	[self notifyDelegateOfStartup];
+	
+	// defer the end of the startup until the main runloop is in action - this gives the client time to add a delegate
+	dispatch_async(dispatch_get_main_queue(), ^{
+		[self finishStartup];
+	});
 }
 
```

Comment:
This still seems like a hack to me. Does the delegate ever change? My preferred pattern when this comes up is `initWithDelegate:`